File: Source/Aladdin.CAPI.STB/ASN1/STB/SBlock.cs
using System; 
using System.Collections.Generic; 

namespace Aladdin.ASN1.STB
{
    ///////////////////////////////////////////////////////////////////////////////
    // SBlockTable ::= OCTET STRING (SIZE(64))
    // SBlock ::= CHOICE {
    // 	table SBlockTable,
    // 	oid OBJECT IDENTIFIER
    // }
    ///////////////////////////////////////////////////////////////////////////////
    public class SBlock : Choice
    {
	    // информация о структуре
	    private static readonly ObjectInfo[] info = new ObjectInfo[] { 

		    new ObjectInfo(new ObjectCreator<OctetString     >().Factory(64, 64), Cast.N), 
		    new ObjectInfo(new ObjectCreator<ObjectIdentifier>().Factory(      ), Cast.N), 
	    }; 
	    // конструктор
	    public SBlock() : base(info) {} 
    }
}

File: Source/Aladdin.CAPI.STB.Java/src/aladdin/asn1/stb/SBlock.java
package aladdin.asn1.stb;
import aladdin.asn1.*; 

///////////////////////////////////////////////////////////////////////////////
// SBlockTable ::= OCTET STRING (SIZE(64))
// SBlock ::= CHOICE {
// 	table SBlockTable,
// 	oid OBJECT IDENTIFIER
// }
///////////////////////////////////////////////////////////////////////////////
public final class SBlock extends Choice
{
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ObjectCreator(OctetString     .class).factory(64, 64), Cast.N), 
		new ObjectInfo(new ObjectCreator(ObjectIdentifier.class).factory(      ), Cast.N), 
	}; 
	// конструктор
	public SBlock() { super(info); } 
}
